src/render/shaders/Shadow.hpp

```diff
@@ -66,17 +66,17 @@ void main() {
         float distanceL = pixCoord[0];
         float distanceR = fullSize[0] - pixCoord[0];
 
-        // get the smallest
-        float smallest = min(min(distanceT, distanceB), min(distanceL, distanceR));
+        // get the distance to the closest edge
+        float distanceToEdge = range - min(min(distanceT, distanceB), min(distanceL, distanceR));
 
-        if (smallest < range) {
-            pixColor[3] = pixColor[3] * pow((smallest / range), shadowPower);
+        if (distanceToEdge > 0.0) {
+            pixColor[3] *= 1.0 - pow(distanceToEdge / range, shadowPower);
```

Comment:
isn't a floating power WAY slower than an integer one?

Reply:
That's fine, a graphics card is expert in FLOPs. But technically it could be a tad slower.

Reply:
I'm not shure to be honest.

Reply:
a non-integer power requires roots, which are notoriously slow.

Reply:
But test the thing, it works beautifully and you have super nice control over the shadow.

Reply:
How about the distance function that you use everywhere?
Honestly hardware nowadays is quite good at roots.

Reply:
you're right, maybe I am a bit too paranoid here. I am stuck in the intel 8086 days.

Reply:
But distance is a squareroot too.
I can optimize those away though, because you don't want the distance but just a distance comparison.
Didn't think about that, i will do that later.
But this should be fine, especially beccause its just done for a small part of the whole texture.

Reply:
You should focus on that alleged 10x performance boost, as I still cannot believe you it would make a difference over 10% :P

Reply:
and I should do some profiling to figure out those constant CPU usage issues